app/scripts/metamask-controller.js

```diff
@@ -2052,6 +1941,12 @@ export default class MetamaskController extends EventEmitter {
       controllersByName.MultichainNetworkController;
     this.authenticationController = controllersByName.AuthenticationController;
     this.userStorageController = controllersByName.UserStorageController;
+    this.notificationServicesController =
+      controllersByName.NotificationServicesController;
+    this.notificationServicesPushController =
+      controllersByName.NotificationServicesPushController;
+
+    this.notificationServicesController.init();
```

Comment:
As discussed in [this core PR](https://github.com/MetaMask/core/pull/5504). We have decoupled the constructor from initialisation.
^ This was because previously we called messenger actions inside the constructor.